.github/workflows/flexci_dispatcher.py

```diff
@@ -0,0 +1,163 @@
+        data=payload,
+        headers={
+            'User-Agent': 'FlexCI-Dispatcher',
+            'Content-Type': 'application/json',
+            'X-GitHub-Event': 'issue_comment',
+            'X-Hub-Signature': 'sha1={}'.format(
+                hmac.new(secret.encode(), payload, 'sha1').hexdigest()),
+            'X-Hub-Signature-256': 'sha256={}'.format(
+                hmac.new(secret.encode(), payload, 'sha256').hexdigest()),
+        },
+    )
+    with urllib.request.urlopen(req) as res:
+        response = json.loads(res.read())
+    if 'job_ids' in response:
+        for job in response['job_ids']:
+            _log(f'Triggered: {base_url}/r/job/{job["id"]}')
+        return True
+    elif 'message' in response:
+        _log(f'Failed to submit webhook payload: {response["message"]}')
+        return False
+    raise RuntimeError('unexpected response: {response}')
+
+
+def _complement_commit_status(
+        repo: str, pull_req: int, token: Optional[str],
+        projects: Set[str], context_prefix: str) -> None:
+    gh_repo = github.Github(token).get_repo(repo)
+    gh_commit = gh_repo.get_commit(gh_repo.get_pull(pull_req).head.sha)
+    _log(f'Checking statuses: {repo}, PR #{pull_req}, commit {gh_commit.sha}')
+    contexts = [s.context for s in gh_commit.get_statuses()]
+    for prj in projects:
+        context = f'{context_prefix}/{prj}'
+        if context in contexts:
+            # Preserve status set via previous (real) CI run.
+            continue
+        _log(f'Setting status as skipped: {context}')
+        gh_commit.create_status(
+            state='success', description='Skipped', context=context)
+
+
+def extract_requested_tags(comment: str) -> Optional[Set[str]]:
+    """
+    Returns the list of test tags requested in the comment.
+    """
+    for line in comment.splitlines():
+        match = re.fullmatch(r'/test ([\w,\- ]+)', line)
+        if match is not None:
+            return set([x.strip() for x in match.group(1).split(',')])
+    return None
+
+
+def parse_args(argv: Any) -> Any:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--webhook', type=str, required=True,
+        help='Path to the JSON file containing issue_comment webhook payload')
+    parser.add_argument(
+        '--projects', type=str, required=True,
+        help='Path to the JSON file containing map from FlexCI project to '
+             'list of tags')
+    parser.add_argument(
+        '--flexci-uri', type=str, default='https://ci.preferred.jp',
+        help='Base URI of the FlexCI server (default: %(default)s)')
+    parser.add_argument(
+        '--flexci-context', type=str, default='pfn-public-ci',
+        help='Context prefix of the FlexCI server (default: %(default)s)')
+    return parser.parse_args(argv[1:])
+
+
+def main(argv: Any) -> int:
+    options = parse_args(argv)
+    webhook_secret = str(os.environ.get('FLEXCI_WEBHOOK_SECRET'))
+    github_token = str(os.environ.get('GITHUB_TOKEN'))
```

Comment:
If these env variables are required,
```suggestion
    webhook_secret = os.environ['FLEXCI_WEBHOOK_SECRET']
    github_token = os.environ['GITHUB_TOKEN']
```